### PR TITLE
Type annotate fastkml.data

### LINKED ISSUE #189: Type annotate fastkml.data
Correct and improve the type annotations of the file `fastkml/data.py` so that the `pyproject.toml` directive to exclude this file from `mypy` can be dropped

### PR DESCRIPTION
**Pull request by ramonsaraiva**
Adds type annotations for data module and includes data module in `pyproject.toml` mypy settings.

Some warnings needed to be ignored because of how mypy deals with getters/setters annotations (see https://github.com/python/mypy/issues/3004)

Closes #189

### CODE REVIEW

File: fastkml/data.py [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
 from typing import Optional
 from typing import Tuple
 from typing import Union
 
 from typing_extensions import TypedDict
 
-import fastkml.config as config
 from fastkml.base import _BaseObject
 from fastkml.base import _XMLObject
+from fastkml.config import etree  # type: ignore[attr-defined]
```
Comment (cleder) [inline fastkml/data.py:11]: Please revert this change, otherwise the user is not able to change the etree implementation to use.

File: fastkml/data.py [hunk 2]
```diff
@@ -1,20 +1,32 @@
...
 
 
 class SimpleField(TypedDict):
     name: str
     type: str
-    displayName: str  # noqa: N815
+    displayName: Optional[str]  # noqa: N815
+
+
+SimpleFields = List[Dict[str, Union[str, str]]]
```
Comment (cleder) [inline fastkml/data.py:21]: `Union[str, str]` does not make any sense to me

Comment (ramonsaraiva) [reply fastkml/data.py:21]: Typo most likely, will fix.

File: fastkml/data.py (hunk 1 above)
Comment (ramonsaraiva) [reply fastkml/data.py:11]: Was trying to solve for `fastkml/data.py:148: error: Module has no attribute "etree"  [attr-defined]` - any suggestions for that?

Comment (ramonsaraiva) [reply fastkml/data.py:11]: Those happen everytime we have a `config.etree` mention:
```
fastkml/data.py:148: error: Module has no attribute "etree"  [attr-defined]
fastkml/data.py:152: error: Module has no attribute "etree"  [attr-defined]
fastkml/data.py:178: error: Module has no attribute "etree"  [attr-defined]
fastkml/data.py:181: error: Module has no attribute "etree"  [attr-defined]
fastkml/data.py:302: error: Module has no attribute "etree"  [attr-defined]
```
I.e.:
`sf = config.etree.SubElement(element, f"{self.ns}SimpleField")`

Comment (cleder) [reply fastkml/data.py:11]: Reverting this will also make the failing tests pass

File: fastkml/data.py [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
+SimpleFieldsInput = Optional[
```
Comment (cleder) [inline fastkml/data.py:22]: This type annotation looks too complex to me. Please try to define smaller subtypes and name them according to their use.
Also consider using [`@overload`](https://adamj.eu/tech/2021/05/29/python-type-hints-how-to-use-overload/) instead to narrow the types down depending on use.

**Review by cleder — CHANGES_REQUESTED** [review]

File: fastkml/data.py (hunk 3 above)
Comment (ramonsaraiva) [reply fastkml/data.py:22]: I agree, but that's a reflection of accepting a list, a dict, or a list of dicts in the same setter. `@overload` might help, but maybe a cleaner implementation would be using `singledispatchmethod`, but that's not available before python 3.8.
